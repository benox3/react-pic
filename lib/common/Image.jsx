import React, { Component, PropTypes } from 'react';

export default class Image extends Component {
  static propTypes = {
    imgStyle: PropTypes.objectOf(
      PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
    ).isRequired,
    isBlurred: PropTypes.bool.isRequired,
    blurAmount: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]).isRequired,
  };

  getStyle() {
    const { imgStyle, isBlurred, blurAmount } = this.props;
    console.log(isBlurred, blurAmount);
    if (isBlurred) {
      return {
        ...imgStyle,
        filter: `blur(${blurAmount})`,
      };
    }

    return imgStyle;
  }

  render() {
    const { alt, image } = this.props;
    return (
      <img
        alt={alt}
        style={this.getStyle()}
        src={image.url}
      />
    );
  }
}
