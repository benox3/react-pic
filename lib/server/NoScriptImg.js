import React, { Component } from 'react';
import convertReactToHTMLStyle from '../utils/convertReactToHTMLStyle';

export default class NoScriptImg extends Component {
  render() {
    const { alt, style, image} = this.props;

    return (
      <noscript dangerouslySetInnerHTML={{
        __html:
          '<img ' +
            `alt='${alt}' ` +
            `style='${convertReactToHTMLStyle(style)}' ` +
            `src='${image.url}' />`
      }} />
    );
  }
}
