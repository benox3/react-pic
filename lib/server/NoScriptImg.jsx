import React, { PropTypes } from 'react';
import convertReactToHTMLStyle from '../utils/convertReactToHTMLStyle';

const NoScriptImg = ({ alt, style, image }) => (
  <noscript
    dangerouslySetInnerHTML={{
      __html:
        '<img ' +
          `alt='${alt}' ` +
          `style='${convertReactToHTMLStyle(style)}' ` +
          `src='${image.url}' />`,
    }}
  />
);

NoScriptImg.propTypes = {
  alt: PropTypes.string.isRequired,
  style: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
  ).isRequired,
  image: PropTypes.shape({
    url: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
  }).isRequired,
};

export default NoScriptImg;
