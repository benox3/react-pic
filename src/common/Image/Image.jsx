import React, { Component, PropTypes } from 'react';

export default class Image extends Component {
  static propTypes = {
    alt: PropTypes.string.isRequired,
    image: PropTypes.shape({
      url: PropTypes.string.isRequired,
      width: PropTypes.number.isRequired,
    }).isRequired,
    imgStyle: PropTypes.objectOf(
      PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
    ).isRequired,
    isBlurred: PropTypes.bool.isRequired,
    blurAmount: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]).isRequired,
  };

  getStyle() {
    const { imgStyle, isBlurred, blurAmount } = this.props;
    if (isBlurred) {
      return {
        ...imgStyle,
        filter: `blur(${blurAmount})`,
      };
    }

    return imgStyle;
  }

  render() {
    const { alt, image } = this.props;
    return (
      <img
        alt={alt}
        style={this.getStyle()}
        src={image.url}
      />
    );
  }
}
