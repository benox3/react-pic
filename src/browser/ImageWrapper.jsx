import React from 'react';
import { Image } from '../common/Image';

const ImageWrapper = props => (
  <Image {...props} />
);

export default ImageWrapper;
