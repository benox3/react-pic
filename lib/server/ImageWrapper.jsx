import React, { PropTypes } from 'react';
import NoScriptImg from './NoScriptImg';
import Image from '../common/Image';

const ImageWrapper = (props) => {
  const { alt, imgStyle, noScriptImage } = props;
  return (
    <div>
      <NoScriptImg
        alt={alt}
        image={noScriptImage}
        style={{
          position: 'absolute',
          zIndex: imgStyle.zIndex ? imgStyle.zIndex + 1 : 10,
          ...imgStyle,
        }}
      />
      <Image {...props} />
    </div>
  );
};

ImageWrapper.propTypes = {
  alt: PropTypes.string,
  imgStyle: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
  ).isRequired,
  noScriptImage: PropTypes.shape({
    url: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
  }).isRequired,
};

ImageWrapper.defaultProps = {
  alt: null,
  noScriptImage: null,
};

export default ImageWrapper;
