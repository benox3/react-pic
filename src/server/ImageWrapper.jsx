import React, { PropTypes } from 'react';
import NoScriptImg from './NoScriptImg';
import Image from '../common/Image';

const ImageWrapper = (props) => {
  const { alt, imgStyle, noscriptImage } = props;
  return (
    <div>
      <NoScriptImg
        alt={alt}
        image={noscriptImage}
        style={{
          position: 'absolute',
          zIndex: imgStyle.zIndex ? imgStyle.zIndex + 1 : 10,
          ...imgStyle,
        }}
      />
      <Image {...props} />
    </div>
  );
};

ImageWrapper.propTypes = {
  alt: PropTypes.string,
  imgStyle: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
  ).isRequired,
  noscriptImage: PropTypes.shape({
    url: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
  }).isRequired,
};

ImageWrapper.defaultProps = {
  alt: null,
  noscriptImage: null,
};

export default ImageWrapper;
