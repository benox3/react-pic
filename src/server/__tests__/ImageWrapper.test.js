import React from 'react';

import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ImageWrapper from '../ImageWrapper';
import NoScriptImg from '../NoScriptImg';
import Image from '../../common/Image';

describe('ImageWrapper', () => {
  const props = {
    alt: 'heart',
    noscriptImage: {
      width: 290,
      url: 'http://placehold.it/290?text=♥',
    },
    image: {
      width: 290,
      url: 'http://placehold.it/290?text=♥',
    },
    imgStyle: {
      margin: 0
    },
    isBlurred: false,
    blurAmount: 0
  };

  describe('rendering', () => {
    it('should render', () => {
      expect(mount(<ImageWrapper {...props} />).html()).to.not.equal(null);
    });

    it('should render a NoScriptImg component', () => {
      expect(
        mount(<ImageWrapper {...props} />).find(NoScriptImg).exists()
      ).to.be.true;
    });

    it('should render a Image component', () => {
      expect(
        mount(<ImageWrapper {...props} />).find(Image).exists()
      ).to.be.true;
    });
  });

  describe('snapshots', () => {
    it('should match base snapshot', () => {
      expect(renderer.create(<ImageWrapper {...props} />)).toMatchSnapshot();
    });
  });
});