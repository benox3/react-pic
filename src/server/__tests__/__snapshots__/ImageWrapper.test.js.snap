// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageWrapper snapshots should match base snapshot 1`] = `
<div>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<img alt='heart' style='position:absolute;z-index:10;margin:0;' src='http://placehold.it/290?text=♥' />",
      }
    }
  />
  <img
    alt="heart"
    src="http://placehold.it/290?text=♥"
    style={
      Object {
        "margin": 0,
      }
    }
  />
</div>
`;
